File: Perl/Enum/src/sample1.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

use constant {
    Red => 'The color is red.',
    Orange => 'The color is orange.',
    Yellow => 'The color is yellow.',
    Green => 'The color is green.',
    Cyan => 'The color is cyan.',
    Blue => 'The color is blue.',
    Purple => 'The color is purple.'
};

sub main {
    print(Blue."\n");
}

main;

File: PHP/Enum/src/sample1.php
<?php

class Color extends SplEnum {
    const RED = "The color is red.";
    const ORANGE = "The color is orange.";
    const YELLOW = "The color is yellow.";
    const GREEN = "The color is green.";
    const CYAN = "The color is cyan.";
    const BLUE = "The color is blue.";
    const PURPLE = "The color is purple.";
}

function main() {
    echo Color::BLUE;
}

main();
